frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -215,7 +215,38 @@ function ReaderBookmark:gotoBookmark(pn_or_xp)
     end
 end
 
+-- This function adds "chapter" property to highlights already saved in the document
+function ReaderBookmark:updateHighlightsIfNeeded()
+    local version = self.ui.doc_settings:readSetting("bookmarks_version") or 0
+    if version >= 20200511 then
+        return
+    end
+
+    for _, highlights in pairs(self.view.highlight.saved) do
+        for _, highlight in pairs(highlights) do
+            local pg_or_xp = self.ui.document.info.has_pages and
+                    highlight.pos0.page or highlight.pos0
```

Comment:
I don't know much about PDF, but can you check how this works with PDFs ?
It looks to me `highlight.pos0.page` does not exist. When highlighting in a PDF, I get saved:
```lua
    ["highlight"] = {
        [1] = {
            [1] = {
                ["drawer"] = "lighten",
                ["datetime"] = "2020-05-24 16:57:39",
                ["pboxes"] = {
                    [1] = {
                        ["y"] = 887,
                        ["x"] = 112,
                        ["h"] = 33,
                        ["w"] = 525
                    }
                },
                ["pos0"] = {
                    ["y"] = 891.06819958378,
                    ["x"] = 388.23636363636
                },
                ["pos1"] = {
                    ["y"] = 891.06819958378,
                    ["x"] = 388.23636363636
                },
                ["text"] = "blah"
            }
        }
    },
```
The page looks like it's actually the key `k` from `for k, v in pairs(self.view.highlight.saved) do`.

(The only other place I found that would use pos0.page is in `ReaderBookmark:isBookmarkSame()`, where it serves nothing but won't cause harm as `nil == nil` :)